src/Executable.js

```diff
@@ -595,6 +629,19 @@ export default class Executable {
                 );
             }
 
+            if (this.transactionNodeAccountIds?.length) {
```

Comment:
if we have the empty array return this check wouldn't be needed and the code would be more "flat".

Reply:
Even if the array is empty, its length must still be checked because `isNodeAccountIdValid` will always evaluate to false. This is because the chosen node for the transaction cannot be found in an empty array.